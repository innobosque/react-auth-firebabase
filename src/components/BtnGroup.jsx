import { useForm } from "../hooks/useForms"

export const BtnGroup = ({ userBool, loading, handleSignInWithEmailPassword, handleSignOut }) => {
    const {onInputChange,email,password} = useForm({
        email: '',
        password: '',
    });
    return (
        <div className="btn-group">
            {
            !userBool?
            <div>
                <input
                className="block w-full my-2 px-2 py-1 rounded" 
                name="email"
                onChange={onInputChange}
                value={email}
                type="text" placeholder="Su correo" />
                <input
                className="block w-full my-2 px-2 py-1 rounded"
                name="password"
                onChange={onInputChange}
                value={password}
                 type="password" placeholder="Su contraseÃ±a" />
                <button
                    aria-label="Acceso"
                    onClick={() => handleSignInWithEmailPassword(email,password)}
                    className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full"
                    style={{ backgroundColor: loading ? 'rgb(248,113,113)' : '' }}>
                    {loading ? <span role="alert">Cargando...</span> : 'Acceso'}
                </button>
            </div>
            :
            <button
                aria-label="Logout"
                onClick={() => handleSignOut()}
                className="py-2 px-4 bg-red-500 hover:bg-red-700 text-white rounded-full">
                Logout
            </button>
            }
        </div>
    )
}
