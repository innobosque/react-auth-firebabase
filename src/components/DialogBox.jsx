import PropTypes from 'prop-types';

export const DialogBox = ({message}) => {
  return (
    <div className="my-2 p-2 text-white bg-red-300 rounded">{message}</div>
  )
}

DialogBox.propTypes = {
  message: PropTypes.string.isRequired
}