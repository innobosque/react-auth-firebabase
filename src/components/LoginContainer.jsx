import React from 'react'

export const LoginContainer = ({ children, title = 'Firebase Authentication'}) => {
    return (
        <div className="bg-slate-500 p-3 rounded">
            <h1 className="py-3 text-xl text-white font-semibold">{title}</h1>
            {children}
        </div>
    )
}
