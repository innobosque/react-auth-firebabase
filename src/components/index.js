export * from './BtnGroup';
export * from './DialogBox';
export * from './LoginContainer';