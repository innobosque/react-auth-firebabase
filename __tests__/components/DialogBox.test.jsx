import { render,screen } from "@testing-library/react";
import { DialogBox } from "../../src/components/DialogBox";

describe('Pruebas sobre el componente DialogBox', () => {
  const message = "Hi Xurxo";

  test("Se debe hacer un match con el snapshot del componente",()=>{
      const {container} = render(<DialogBox message={message} />);
      expect(container).toMatchSnapshot();
  })

  test(`Se debe encontar en el componente el texto "${message}"`,()=>{
    //Sin screen
    //const {getByTestId} = render(<DialogBox message={message} />);
    //expect(getByTestId('dialog-box').innerHTML).toContain(message);
    render(<DialogBox message={message} />);
    expect(screen.getByTestId('dialog-box').innerHTML).toBeTruthy();
    expect(screen.getByTestId('dialog-box').innerHTML).toContain(message);
  });

  test('Que el texto buscado se encuentre en un elemento "div"',()=>{
    render(<DialogBox message={message} />);
    const label = screen.getByRole("alert");
    expect(label).toBeTruthy();
    expect(label.nodeName).toBe("DIV");
  });

 


})