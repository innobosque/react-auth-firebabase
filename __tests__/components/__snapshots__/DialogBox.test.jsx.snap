// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <DialogBox/ > Se debe hacer un match con el snapshot del componente 1`] = `
<div>
  <div
    class="dialog-box my-2 p-2 text-white bg-red-300 rounded"
    data-testid="dialog-box"
    role="alert"
  >
    Hi Xurxo
  </div>
</div>
`;
